index.js

```diff
@@ -1016,41 +1020,63 @@ const createFocusTrap = function (elements, userOptions) {
       return this;
     },
 
+    _isManuallyPaused() {
+      return state.manuallyPaused;
+    },
+
+    _setPausedState(paused, options) {
+      state.paused = paused;
+      if (paused) {
+        const onPause = getOption(options, 'onPause');
+        const onPostPause = getOption(options, 'onPostPause');
+        onPause?.();
+
+        removeListeners();
+        updateObservedNodes();
+
+        onPostPause?.();
+      } else {
+        const onUnpause = getOption(options, 'onUnpause');
+        const onPostUnpause = getOption(options, 'onPostUnpause');
+
+        onUnpause?.();
+
+        updateTabbableNodes();
+        addListeners();
+        updateObservedNodes();
+
+        onPostUnpause?.();
+      }
+
+      return this;
+    },
+
     pause(pauseOptions) {
-      if (state.paused || !state.active) {
+      if (!state.active) {
         return this;
       }
 
-      const onPause = getOption(pauseOptions, 'onPause');
-      const onPostPause = getOption(pauseOptions, 'onPostPause');
-
-      state.paused = true;
-      onPause?.();
+      state.manuallyPaused = true;
 
-      removeListeners();
-      updateObservedNodes();
+      if (state.paused) {
+        return this;
+      }
```

Comment:
Not necessary if the state check moves into `_setPausedState()` (see previous comment).

Reply:
Done